GitVersionCore.Tests/GitVersionContextTests.cs

```diff
@@ -50,5 +51,26 @@ public void UsesBranchSpecificConfigOverTopLevelDefaults()
             context.Configuration.VersioningMode.ShouldBe(VersioningMode.ContinuousDeployment);
             context.Configuration.Tag.ShouldBe("alpha");
         }
+
+        [Test]
+        public void CanFindParentBranchForInheritingIncrementStrategy()
+        {
+            var config = new Config();
+            config.Branches["develop"].Increment = IncrementStrategy.Major;
+            config.Branches["feature[/-]"].Increment = IncrementStrategy.Inherit;
```

Comment:
This solves the issue with feature branches which are taken off master should bump patch but feature branches off develop should bump minor